### PR TITLE
LABS-1200 Fix retries in block manager and emulate materialization delays in PALM for disagg storage

### PR DESCRIPTION
**Pull request by ddanderson**
- Fixed retries in block manager, the result_count was not being reset.
- Added a way, with a palm config variable, to emulate materialization delays
- Added a test to confirm that it all works

### CODE REVIEW

**Comment by ddanderson** [comment]
@pmacko86, could you please review?

File: ext/page_log/palm/palm.c [hunk 1]
```diff
@@ -333,6 +333,24 @@ palm_resize_item(WT_ITEM *item, size_t new_size)
...
+{
+    memset(context, 0, sizeof(*context));
+
+    /*
+     * To get more testing variation, We could call palm_compute_delay_us to randomize this number.
+     * If we do so, we need to make sure items are materialized in the same order they are written.
+     * So when setting PAGE_KEY.timestamp_materialized_us, we'd need to make each value set was
+     * monotonically increasing.
+     */
+    context->materialization_delay_us = palm->materialization_delay_ms;
```
Comment (pmacko86) [inline ext/page_log/palm/palm.c:351]: "us" and "ms" are different units... did you forget to multiply by 1000?

File: test/suite/helper_disagg.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def conn_extensions(self, extlist):
     #
     # Some possible values to return: 'verbose=1'
     # or for palm: 'verbose=1,delay_ms=13,force_delay=30'
+    # or 'materialization_delay_ms=1000000'
```
Comment (pmacko86) [inline test/suite/helper_disagg.py:101]: 1000 seconds is a long delay...

File: test/suite/test_oligarch10.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, time, wiredtiger, wttest
+
+# test_oligarch10.py
+#    Create an artificial delay in materializing pages from the page service.
+class test_oligarch10(wttest.WiredTigerTestCase):
```
Comment (pmacko86) [inline test/suite/test_oligarch10.py:33]: I added `test_oligarch10` already, so please rename this to 11

File: test/suite/test_oligarch10.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                + 'oligarch=(role="leader"),' \
+                + 'disaggregated=(stable_prefix=.,page_log=palm),'
+
+    uri = "oligarch:" + uri_base
+
+    # Load the directory store extension, which has object storage support
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        config='materialization_delay_ms=1000000'
```
Comment (pmacko86) [inline test/suite/test_oligarch10.py:47]: 1000 seconds?

**Review by pmacko86 — APPROVED** [review]
LGTM. It just looks like you have been using "us" and "ms" interchangeably. I like the idea of using "ms" as the unit in the configuration, even if you keep "us" internally.

File: test/suite/test_oligarch10.py (hunk 4 above)
Comment (ddanderson) [reply test/suite/test_oligarch10.py:47]: I'll fix, I kept adding zeros, and didn't realize there was a unit error (kind of in a rush for testing).

File: ext/page_log/palm/palm.c (hunk 1 above)
Comment (ddanderson) [reply ext/page_log/palm/palm.c:351]: Ah - that's it, I'll fix.

**Pull request merged**